server/index.js

```diff
@@ -3,13 +3,30 @@ import { graphqlExpress, graphiqlExpress } from 'graphql-server-express';
 import { createServer } from 'http';
 import { SubscriptionServer } from 'subscriptions-transport-ws';
 import bodyParser from 'body-parser';
+import { makeExecutableSchema } from 'graphql-tools';
 
-import schema from '../schema';
+import typeDefs from '../schema';
+import resolvers from '../resolvers';
 import addModelsToContext from '../models';
 
 import { pubsub, subscriptionManager } from './subscriptions';
 import connectToMongo from './mongo';
 
+// XXX: this is a total hack, better to use graphql-tools for this bit
+['Query', 'Mutation', 'Subscription'].forEach((rootField) => {
+  const parts = [];
+  const re = new RegExp(`type ${rootField}[\\s\\n]*{([^}]*)}`);
+  typeDefs.forEach((subschema) => {
+    const match = re.exec(subschema);
+    if (match) {
+      parts.push(match[1].trim());
+    }
+  });
+  typeDefs.push(`type ${rootField} { ${parts.join('\n')} }\n\n`);
+});
```

Comment:
A non-hacked version of this that uses ASTs will move to graphql-tools (as `extend type`)